ddtrace/_trace/tracer.py

```diff
@@ -775,8 +775,7 @@ def _start_span(
         service = config.service_mapping.get(service, service)
 
         links = context._span_links if not parent else []
-
-        if trace_id:
+        if trace_id or links:
```

Comment:
Should this also consider baggage here? ie: we should propagate a context if it has trace info, span links, and/or baggage. A comment would be useful here too. 

```suggestion
        if trace_id or links or context._baggage:
```

Reply:
Also including links here seems like a separate bug fix, should we call this out in a release note?

Reply:
I think previously we never would've gotten here with a context that just had a span link via propagation. However I guess you're saying in case users are trying to create a span from a context with just a span link manually? I think that this is such an obscure edge case that it doesn't need a release note but if you disagree I can add.